Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -0,0 +1,223 @@
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsObject('iconContext', iconContext);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckTypeIsFunction('onFailure', onFailure);
+
+    var newCommand = BocReferenceValueBase.CreateCommand(oldCommand, commandInfo, businessObject);
+
+    var oldIcon = oldCommand.find('img');
+    var newIcon = BocReferenceValueBase.CreateEmptyIcon(oldIcon, newCommand.prop('title'));
+    newCommand.append(newIcon);
+
+    oldCommand.replaceWith(newCommand);
+
+    if (isIconUpdateEnabled)
+    {
+      var pageRequestManager = Sys.WebForms.PageRequestManager.getInstance();
+      if (pageRequestManager.get_isInAsyncPostBack())
+        return oldCommand;
+
+      var params: any = { businessObject: businessObject };
+      for (var propertyName in iconContext)
+        params[propertyName] = (iconContext as any)[propertyName];
+
+      Sys.Net.WebServiceProxy.invoke(
+          controlServiceUrl!,
+          'GetIcon',
+          false,
+          params,
+          function (result: string, eventArgs: Sys.EventArgs): void
+          {
+            BocReferenceValueBase.UpdateIconFromWebService(newCommand, newIcon, JSON.parse(result)); //TODO event handler signature requires string but 'UpdateIconFromWebService' obviously requires some other objecct
+          },
+          function (err)
+          {
+            onFailure(err);
+            BocReferenceValueBase.ResetCommand(newCommand);
+          });
+    }
+
+    return newCommand;
+  };
+
+  public static CreateCommand (oldCommand: JQuery, commandInfo: Nullable<CommandInfo>, businessObject: any): JQuery
+  {
+    ArgumentUtility.CheckNotNull('oldCommand', oldCommand);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckTypeIsString('businessObject', businessObject);
+
+    var newCommand = $('<a/>');
+
+    var tempCommandInfo: CommandInfo = {
+      href: null,
+      target: null,
+      onClick: null,
+      title: null
+    }
+    if (commandInfo != null)
+    {
+      tempCommandInfo = jQuery.extend(true, {}, commandInfo);
+      if (businessObject == null)
+      {
+        tempCommandInfo.href = null;
+        tempCommandInfo.target = null;
+        tempCommandInfo.onClick = null;
+        tempCommandInfo.title = null;
+      }
+      else if (commandInfo.href != null)
+      {
+        tempCommandInfo.href = String.format(commandInfo.href, businessObject);
+      }
+    }
+
+    var oldCommandAttributes = oldCommand[0].attributes;
+
+    for (var i = 0; i < oldCommandAttributes.length; i++)
+    {
+      var value = oldCommandAttributes[i].nodeValue;
+      if (value != null && value != '')
+        newCommand.attr(oldCommandAttributes[i].nodeName, value);
+    }
+
+    for (var property in tempCommandInfo)
+    {
+      var value: Nullable<string> = "" + (tempCommandInfo as any)[property]; // TODO is there a better way to do this?
```

Comment:
@a-ctor see todo, but given we're going to be dropping the command, we shouldn't invest here.

also, single quotes instead of double quotes.

Reply:
`tempCommandInfo` should have the type `Dictionary<unknown>`, allowing arbitrary properties to exist on the object. Since the property is not actually limited to the four properties that are currently assigned.